browser/brave_content_browser_client.cc

```diff
@@ -220,11 +225,34 @@ void BindBraveSearchHost(
 
   content::BrowserContext* context = render_process_host->GetBrowserContext();
   mojo::MakeSelfOwnedReceiver(
-      std::make_unique<brave_search::BraveSearchHost>(
+      std::make_unique<brave_search::BraveSearchFallbackHost>(
           content::BrowserContext::GetDefaultStoragePartition(context)
               ->GetURLLoaderFactoryForBrowserProcess()),
       std::move(receiver));
 }
+
+void BindBraveSearchDefaultHost(
+    content::RenderFrameHost* const frame_host,
+    mojo::PendingReceiver<brave_search::mojom::BraveSearchDefault> receiver) {
+  auto* context = frame_host->GetBrowserContext();
+  auto* profile = Profile::FromBrowserContext(context);
+  auto* template_url_service =
+      TemplateURLServiceFactory::GetForProfile(profile);
+  const std::string host = frame_host->GetLastCommittedURL().host();
+  if (brave::IsRegularProfile(profile)) {
+    mojo::MakeSelfOwnedReceiver(
+        std::make_unique<brave_search::BraveSearchDefaultHost>(
+            host, template_url_service, profile->GetPrefs()),
+        std::move(receiver));
+  } else {
+    // Dummy API which always returns false for private contexts.
+    mojo::MakeSelfOwnedReceiver(
+        std::make_unique<brave_search::BraveSearchDefaultHostPrivate>(
```

Comment:
do we actually need a different class for this? Can't we just return early if these values are null? Or alternatively why pass in the values at all if they don't matter?

Reply:
Well I just thought it might be a clean way to keep the private profile implementation separate from the regular profile, since there is no code overlap. Even if we allow the setting in private profile in the future, we'd have to use something similar to `kUseAlternativeSearchEngineProvider`. We need to pass in values because it inherits from `brave_search::mojom::BraveSearchDefault`, but I suppose I can provide an empty constructor in the subclass.